### PR TITLE
feat: check the remaining before sending the swap transaction

### LINKED ISSUE #2214: [Bug] Should refresh the pool balance before sending swap transaction
### System

- [ ] Windows

  - OS Version:

- [x] Mac OS X

  - OS Version:

- [ ] Linux

  - Linux Distribution:
  - OS Version:

### Platform/Browser

- [x] Chrome

  - Mask Version:
  - Browser Version:

- [ ] Firefox

  - Mask Version:
  - Browser Version:

- [ ] Android

  - Mask Version:
  - Android Version:

- [ ] iOS
  - Mask Version:
  - iOS Version:

### Build Variant

- Where do you get Mask?
  - [x] Store
  - [ ] ZIP
  - [ ] Self-Compiled
    - Build Commit: /_Optionally attach a Commit ID, if it is from an pre-release branch head_/

## Bug Info

The pool balance was fetched when the ITO card was presented. It should refresh before the swap transaction to be sent.

### PR DESCRIPTION
**Pull request by guanbinrui**
close #2214

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 1]
```diff
@@ -68,6 +68,26 @@ export function useClaimCallback(
...
+            }
+        } catch (e) {
+            setClaimState({
+                type: TransactionStateType.FAILED,
+                error: new Error('Failed to check availability.')
+            })
+            return
+        }
+
         const params: Parameters<typeof ITO_Contract['methods']['swap']> = [
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:91]:
```suggestion
        const swapParams: Parameters<typeof ITO_Contract['methods']['swap']> = [
```

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 2]
```diff
@@ -80,7 +100,7 @@ export function useClaimCallback(
             total,
         ]
 
-        // step 1: estimate gas
+        // step 2-1: estimate gas
         const estimatedGas = await ITO_Contract.methods
             .swap(...params)
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:105]:
```suggestion
            .swap(...swapParams)
```

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 3]
```diff
@@ -68,7 +68,27 @@ export function useClaimCallback(
...
-        const params: Parameters<typeof ITO_Contract['methods']['swap']> = [
+        // step 1: check remaining
+        try {
+            const availability = await ITO_Contract.methods.check_availability(id).call({
+                from: account,
+            })
+            if (new BigNumber(availability.remaining).isZero()) {
+                setClaimState({
+                    type: TransactionStateType.FAILED,
+                    error: new Error('Out of Stock'),
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:79]: how about adding it in i18n?

Comment (septs) [reply packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:79]: ??????

Comment (guanbinrui) [reply packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:79]: Maybe in another PR.

**Review by septs — APPROVED** [review]

**Pull request merged**